### PR TITLE
Shell pass

### PR DESCRIPTION
**Pull request by SergeySkoret**

### CODE REVIEW

File: shell/shell.c [hunk 1]
```diff
@@ -1,4 +1,117 @@
...
+#include <stdio.h>
+#include <unistd.h>
+#include <sys/wait.h>
+
+#define ARRAY_SIZE(a) (sizeof(a) / sizeof(*a))
+
+static const struct {
+    const char *app;
+    const char *path;
+} app_list[] = {
```
Comment (AntonKozlov) [inline shell/shell.c]: https://github.com/AntonKozlov/eduos/pull/10#discussion_r140638627 applies here as well

**Comment by AntonKozlov** [comment]
Sorry for the noise, it's accepted as HW, but will require further changes that you, me or someone else could address.

**Comment by SergeySkoret** [comment]
I've completed today's tasks with scheduler and improved my host_shell according to your advice. You can see the results of tests below.
image

But Travis doesn't pass my eduos_shell for the following reason.
How to fix that?
image

**Comment by AntonKozlov** [comment]
as far as I can see, https://github.com/AntonKozlov/eduos/pull/12/commits/3e850e19e5eb0261316b3505dffc4f1226405c85 breaks shell. Sorry, going to take #10

**Pull request closed without merging**